vm/src/obj/objlist.rs

```diff
@@ -153,6 +153,26 @@ pub fn list_extend(vm: &mut VirtualMachine, args: PyFuncArgs) -> PyResult {
     Ok(vm.get_none())
 }
 
+fn list_index(vm: &mut VirtualMachine, args: PyFuncArgs) -> PyResult {
+    trace!("list.index called with: {:?}", args);
+    arg_check!(
+        vm,
+        args,
+        required = [(list, Some(vm.ctx.list_type())), (needle, None)]
+    );
+    for (index, element) in get_elements(list).iter().enumerate() {
+        match vm.call_method(needle, "__eq__", vec![element.clone()]) {
+            Ok(value) => {
+                if objbool::get_value(&value) {
+                    return Ok(vm.context().new_int(index.to_bigint().unwrap()))
+                }
+            }
+            Err(_) => return Err(vm.new_type_error("".to_string())),
+        }
+    }
+    Err(vm.new_value_error(format!("'{:?}' is not in list", needle.borrow())))
```

Comment:
This is the issue I mentioned in the comment.

Reply:
Instead of rust formatting, use the python formatting method by calling `vm.to_str` to call the python `__str__` method instead and retrieve the repr or str value of the object.